nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -0,0 +1,161 @@
+    """
+    site_list: list = []
+    for term in [circuit.termination_a, circuit.termination_z]:
+        if term is not None and getattr(term, "provider_network") is None and term.site not in site_list:
+            site_list.append(term.site)
+
+    return site_list
+
+
+def build_sites_to_maintenance_mapper(maintenance_queryset):
+    """Build a site to circuit maintenance mapper so the data can be quickly accessed of what possible overlaps.
+
+    Leverages defaultdict to provide the default value of an empty set to each key that will be added. Then adds each
+    particular circuit maintenance object to the dictionary to be used as a map of maintenances going on at the site.
+    Build a data dictionary that maps sites to circuit maintenances:
+
+    data_dict = {
+        "msp": {
+            CM1,
+            CM2
+        },
+        "nyc": {
+            CM3,
+            CM4
+        }
+    }
+
+    Args:
+        maintenance_queryset (Queryset): Queryset of all Circuit Maintenance objects
+
+    Returns:
+        dict: Dictionary of a set of maintenance records
+    """
+    return_dictionary = collections.defaultdict(set)
+    for record in maintenance_queryset:
+        # Get the site overall
+        # Start with looping over the circuits that are defined
+        for circuit in record.circuits:
+            # Check both termination A and Z for values of None, add that record to the set for each site
+            for term in [circuit.termination_a, circuit.termination_z]:
+                if term is not None:
+                    return_dictionary[term.site.name].add(record)
+
+    return dict(return_dictionary)
+
+
+class FindSitesWithCircuitImpact(Job):
+    """Nautobot Job definition for finding sites without redundant circuit for impactful maintenance.
+
+    Args:
+        Job (Nautobot Job): Nautobot Job import.
+    """
+
+    class Meta:
+        """Meta definition for the Job."""
+
+        name = "Find Sites With Circuit Impact"
+        description = "Search for sites with multiple circuits, 1 or more circuit impacts."
+        read_only = True
+
+    def run(self, data=None, commit=None):
+        """Executes the Job.
+
+        Args:
+            data (_type_, optional): _description_. Defaults to None.
+            commit (_type_, optional): _description_. Defaults to None.
+        """
+        # Query for all of the circuits maintenances that are on going in the future
+        today = date.today()
+        circuit_maintenances = CircuitMaintenance.objects.filter(start_time__gte=today).order_by("start_time")
+
+        # Build a circuit mapper
+        circuit_maintenance_mapper = build_sites_to_maintenance_mapper(circuit_maintenances)
+
+        # Loop over each of the circuit maintenance records
+        for ckt_maint in circuit_maintenances:
+            # Get the list of sites
+            for circuit in ckt_maint.circuits:
+                sites = get_sites_from_circuit(circuit)
```

Comment:
```suggestion
                sites += get_sites_from_circuit(circuit)
```
and define `sites = []` outside of the loop

Reply:
✅